dataset/get-metadata_test.go

```diff
@@ -55,76 +59,104 @@ func TestUnitHandlers(t *testing.T) {
 	})
 
 	Convey("test getEditMetadataHandler", t, func() {
-		Convey("when working", func() {
-
-			mockDatasetDetails := dataset.DatasetDetails{
-				ID:                "",
-				CollectionID:      "",
-				Contacts:          nil,
-				Description:       "",
-				Keywords:          nil,
-				License:           "",
-				Links:             dataset.Links{},
-				Methodologies:     nil,
-				NationalStatistic: false,
-				NextRelease:       "",
-				Publications:      nil,
-				Publisher:         nil,
-				QMI:               dataset.Publication{},
-				RelatedDatasets:   nil,
-				ReleaseFrequency:  "",
-				State:             "",
-				Theme:             "",
-				Title:             "",
-				UnitOfMeasure:     "",
-				URI:               "",
-				UsageNotes:        nil,
-			}
 
-			mockVersionDetails := dataset.Version{
-				Alerts:        nil,
-				CollectionID:  "",
-				Downloads:     nil,
-				Edition:       "",
-				Dimensions:    nil,
-				ID:            "",
-				InstanceID:    "",
-				LatestChanges: nil,
-				Links:         dataset.Links{},
-				ReleaseDate:   "",
-				State:         "",
-				Temporal:      nil,
-				Version:       0,
-			}
+		mockDatasetDetails := dataset.DatasetDetails{
+			ID:                "test-dataset",
+			CollectionID:      "",
```

Comment:
I was just wandering if these fields need to be set if they are empty?

I think the json.Unmarshal will populate the fields to be empty strings or nil values

Reply:
I just tested this and you wont need to set fields to empty values (either nil, "", false or 0)

Reply:
```
mockDatasetDetails := dataset.DatasetDetails{
			ID:    "test-dataset",
			Links: dataset.Links{LatestVersion: dataset.Link{URL: "/v1/datasets/test/editions/test/version/1"}},
		}

		mockVersionDetails := dataset.Version{
			ID:      "test-version",
			Version: 1,
		}

		mockCollection := zebedee.Collection{
			ID: "test-collection",
			Datasets: []zebedee.CollectionItem{
				{
					ID:    "foo",
					State: "inProgress",
				},
			},
		}
```

Reply:
They were all set and empty from earlier work i think but happy to remove them. Looks cleaner